Assignment 3 Instructions.Rmd

```diff
@@ -153,16 +153,161 @@ K7 <- dplyr::count(K4, cluster)
+# Do the same for 3 clusters:
+classfit_3clu <- kmeans(C4,3)
+C44 <- data.frame(C4,classfit_3clu$cluster)
+names(C44) <- c("3","4","5","6","7","8","9","10","11","12","cluster")
+
+#create visualization
+C44 <- gather(C44, key= "question", value="answer", -cluster)
+C45 <- C44 %>% group_by(question, cluster) 
+C45 <- summarise(C45, avg = mean(answer))
+class(C45$question)
+C45$question <- as.numeric(C45$question)
+class(C43$cluster)
+C45$cluster <- as.factor(C45$cluster)
+
+# get the plot
+ggplot(C45, aes(question, avg, colour = cluster)) + geom_line() + xlab("Question") + ylab("Average Anwer")
+
+```
+Comparing the visualization with 2 clusters and 3 clusters, we found that 2 clusters makes more sense here because the two clusters differ form each other to a great extent while if we divid students into 3 clusters we found there are clusters' answers are similar to each other and there is no need to further divide the third group. Therefore I go for 2 cluster.
+
+``` {r}
+# Create clusters depending on Regions people come from
+View(C5)
+library("ggmap")
+names(C5) <- c("City","Province","Country")
+C51<- geocode(as.character(C5$City),output="latlon")
+row.names(C51)<- c("2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22")
+
+#create clusters depending on latitude and longitudes
+regionfit_2clu <- kmeans(C51,2)
+C51 <- data.frame(C51,regionfit_2clu$cluster)
+names(C51) <- c("lon","lat","cluster")
+
+#create visualization
+C52 <- gather(C51, key= "type", value="lonlan", -cluster)
+C53 <- C52 %>% group_by(type, cluster) 
+C54 <- summarise(C53, avg = mean(lonlan))
+class(C43$cluster)
+C45$cluster <- as.factor(C45$cluster)
+
+# get the plot
+# VISUALIZATION
+```
 ##Part III
 
 Create a visualization that shows the overlap between the two groups of clusters you created in part III.
+``` {r}
+ans_c <- select(C41, 11)
+reg_c <- select(C51,3)
+
+ans_c$reason <- "answers"
+reg_c$reason <- "regions"
+ans_c$student <-rownames(ans_c)
+reg_c$student <-rownames(reg_c)
+
+# plot 1
+C_cluster <- rbind(ans_c, reg_c)
+C_cluster$reason <- as.factor(C_cluster$reason)
+C_cluster$student <- as.numeric(C_cluster$student)
+ggplot(C_cluster, aes(student, cluster, colour = reason)) + geom_point() + xlab("Student") + ylab("Cluster")
+
+# plot 2
+ans <- select(ans_c,1)
+reg <- select(reg_c,1)
+names(ans) <- c("answer")
+names(reg) <- c("region")
+C_cluster_c <- cbind(ans,reg)
+C_cluster_c$overlap <- ifelse(C_cluster_c$answer == C_cluster_c$region,1,0)
+C_cluster_c$student <- row.names(C_cluster_c)
+
+C_sum <- C_cluster_c %>% group_by(overlap) 
+C_sum <- summarise(C_sum, number=n())
+C_sum$number <- as.numeric(C_sum$number)
+C_sum$overlap <- as.factor(C_sum$overlap)
+
+# draw the plot based on overlapping:
+ggplot(C_cluster_c,aes(overlap)) + geom_histogram(aes(y=..count..), binwidth = 0.5)
+ggplot(C_sum,aes(x="",y=number,fill=overlap))+geom_bar(width=1, stat="identity")+coord_polar("y",start = 0)
+```
```

Comment:
These are two interesting solutions to this problem. It may not be the easiest way to visualize the information though. You may have done better with an x/y bar chart or what is called a mosaic plot. You may want to check these out if you have some time.